releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java

```diff
@@ -19,65 +19,136 @@
 
 import org.apache.commons.io.FileUtils;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
+import org.kohsuke.github.GHFileNotFoundException;
 import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHRepository;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
 
-import com.github.checkstyle.CliOptions;
-import com.github.checkstyle.CliOptions.Builder;
+import com.github.checkstyle.Main;
 import com.github.checkstyle.MainProcess;
-import com.github.checkstyle.globals.Constants;
-import com.github.checkstyle.globals.ReleaseNotesMessage;
-import com.google.common.collect.ArrayListMultimap;
-import com.google.common.collect.Multimap;
+import com.github.checkstyle.git.CsGit;
+import com.github.checkstyle.github.CsGitHub;
+import com.github.checkstyle.internal.GhIssueUtil;
+import com.github.checkstyle.internal.RevCommitUtil;
+import com.github.checkstyle.publishers.TwitterPublisher;
 
 public class TemplateProcessorTest {
+    private static final Set<RevCommit> TEST_COMMITS = new LinkedHashSet<>();
+
+    private static final Set<GHIssue> TEST_ISSUES = new LinkedHashSet<>();
+
+    private static final Answer<GHIssue> ISSUE_ANSWER = new Answer<GHIssue>() {
+        @Override
+        public GHIssue answer(InvocationOnMock invocation) throws Throwable {
+            final int findIssue = invocation.getArgument(0);
+
+            for (GHIssue item : TEST_ISSUES) {
+                if (item.getNumber() == findIssue) {
+                    return item;
+                }
+            }
+
+            throw new GHFileNotFoundException("Cannot find Issue: " + findIssue);
+        }
+    };
 
-    private static final String MSG_RELEASE_IS_MINOR =
-        "[ERROR] Validation of release number failed. Release number is minor(1.0.0), but "
-            + "release notes do not contain 'new' or 'breaking compatability' labels. Please "
-            + "correct release number by running https://github.com/checkstyle/checkstyle/"
-            + "actions/workflows/bump-version-and-update-milestone.yml";
-    private static final String MSG_RELEASE_IS_PATCH =
-        "[ERROR] Validation of release number failed. Release number is a patch(1.0.1), but "
-            + "release notes contain 'new' or 'breaking compatability' labels. Please correct "
-            + "release number by running https://github.com/checkstyle/checkstyle/actions/"
-            + "workflows/bump-version-and-update-milestone.yml";
+    private static final MockedStatic<CsGitHub> MOCK_CS_GIT_HUB_STATIC = mockStatic(CsGitHub.class);
+
+    private static final MockedStatic<CsGit> MOCK_CS_GIT_STATIC = mockStatic(CsGit.class);
+
+    private static final MockedStatic<TwitterPublisher>
```

Comment:
These `MockedStatic`s need to be a class field so we can close them after all tests have been run. If not, we get `static mocking is already registered in the current thread To create a new mock, the existing static mock registration must be deregistered` error when tests in `MainTest` start to run.

Reply:
Seems like we got this error in CI as well: https://ci.appveyor.com/project/Checkstyle/contribution/builds/45550414/job/noa9wcc5lknd0b7l#L221
```
For com.github.checkstyle.github.CsGitHub, static mocking is already registered in the current thread
To create a new mock, the existing static mock registration must be deregistered
	at com.github.checkstyle.templates.TemplateProcessorTest.<clinit>(TemplateProcessorTest.java:96)
	... 24 more
```
I think this deregistering has to be done in every test class that uses static mocks.

Reply:
I do not like static. Sharing fields between test methods is always a bad, and problematic in long run.

please look at my extra commit, to avoid static.

please try to convert fields to local variables, the less fields the better.
test should be something like:
```
public void testGenerateOnlyBreakingCompatibility() throws Exception {
commits=....
issues=...
setup(commits, issues)
runMainAndAssertReturnCode(
```

Reply:
unfortunately this is not readable test:
```
        TEST_ISSUES.addAll(createAllIssues(BUG_LABEL));
        TEST_ISSUES.addAll(createAllIssues(MISCELLANEOUS_LABEL));
        TEST_COMMITS.addAll(createAllCommits());

```